applications/feedback_sentiment_analyzer/SPECIFICATION.md

```diff
@@ -0,0 +1,383 @@
+
+For example:
+
+```json
+{
+  "bucket": "my-s3-bucket",
+  "region": "us-east-1",
+  "object": "foo/bar.png"
+}
+```
+
+#### **Output**
+
+Returns a string representing the text extracted.
+
+For example:
+
+```
+CET HÔTEL ÉTAIT SUPER
+```
+---
+### AnalyzeSentiment
+Uses Amazon Comprehend's [DetectSentiment](https://docs.aws.amazon.com/comprehend/latest/APIReference/API_DetectSentiment.html)
+method to detect sentiment (`POSITIVE`, `NEUTRAL`, `MIXED`, or `NEGATIVE`).
+
+#### **Input**
+
+Uses the data available on the [Lambda event object](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-concepts.html#gettingstarted-concepts-event).
+
+For example:
+
+```json
+{
+  "source_text": "CET HÔTEL ÉTAIT SUPER",
+  "region": "us-east-1"
+}
+```
+
+#### **Output**
+
+Returns the guessed sentiment and language code.
+
+For example:
+
+```json
+{
+  "sentiment": "POSITIVE",
+  "language_code": "fr-FR"
+}
+```
+---
+### TranslateText
+Uses Amazon Translate's [TranslateText](https://docs.aws.amazon.com/translate/latest/APIReference/API_TranslateText.html)
+method to translate text to English and return a string.
+
+#### **Input**
+
+Uses the data available on the [Lambda event object](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-concepts.html#gettingstarted-concepts-event).
+
+For example:
+
+```json
+{
+    "source_language_code": "fr-FR",
+    "region": "us-east-1",
+    "extracted_text": "CET HÔTEL ÉTAIT SUPER"
+}
+```
+
+#### **Output**
+
+Returns a string representing the translated text.
+
+For example:
+
+```
+THIS HOTEL WAS GREAT
+```
+---
+### SynthesizeAudio
+Uses Amazon Polly's [SynthesizeAudio](https://docs.aws.amazon.com/polly/latest/dg/API_SynthesizeSpeech.html)
+method to converts input text into life-like speech
+
+#### **Input**
+
+Uses the data available on the [Lambda event object](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-concepts.html#gettingstarted-concepts-event).
+
+For example:
+
+```json
+{
+  "bucket": "my-s3-bucket",
+  "translated_text": "THIS HOTEL WAS GREAT",
+  "region": "us-east-1",
+  "object": "comment.png"
+}
+```
+
+#### **Output**
+
+Returns a string representing the key of the synthesized audio file.
+
+For example:
+
+```
+my-s3-bucket/audio.mp3
+```
+
+---
+
+## GetFeedback Lambda function
+Uses DynamoDB's [GetItem](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_GetItem.html)
+method to get all records from [the table](#managing-items-in-dynamodb). Invoked by the frontend interace.
+
+There is no input.
+
+### **Output**
+
+Returns a JSON object with one property: `feedback`. `feedback` is an array of objects that contain 4 properties:
+- sentiment - "POSITIVE" | "NEGATIVE" | "NEUTRAL"
+- text - The original text translated to the destination language. (English by default)
+- audioUrl - The S3 object key for the synthesized audio file.
+- imageUrl - The original uploaded image.
```

Comment:
```suggestion
Returns a JSON object with one property: `feedback`. `feedback` is an array of objects that contain four properties:
- sentiment - "POSITIVE" | "NEGATIVE" | "NEUTRAL"
- text - The original text translated to the destination language. (English by default.)
- audioUrl - The Amazon S3 object key for the synthesized audio file.
- imageUrl - The original uploaded image.
```

Reply:
Some lists are corrected to have ending punctuation and some are not. What are the guidelines here?